src/plugins/controls/public/control_group/utils/is_compressed.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { apiHasParentApi } from '@kbn/presentation-publishing';
+
+interface HasCompressed {
+  compressed: boolean;
+}
+
+const apiHasCompressed = (unknownApi: unknown): unknownApi is HasCompressed => {
+  return Boolean(unknownApi) && typeof (unknownApi as HasCompressed).compressed === 'boolean';
+};
+
+export function isCompressed(api: unknown): boolean {
+  if (apiHasCompressed(api)) return api.compressed;
+  return apiHasParentApi(api) ? isCompressed(api.parentApi) : true;
+}
```

Comment:
Can we maybe add a unit test for this? 🙇

Reply:
for sure! [b21f593](https://github.com/elastic/kibana/pull/193081/commits/b21f593aa0e10a2dacadd04b4f82cbe98e391547) for scenarios I could think of...lmk what you think